Lib/idlelib/editor.py

```diff
@@ -57,6 +58,15 @@ class EditorWindow(object):
     help_url = None
 
     def __init__(self, flist=None, filename=None, key=None, root=None):
+        from idlelib.autocomplete import AutoComplete
+        from idlelib.autoexpand import AutoExpand
+        from idlelib.calltips import CallTips
+        from idlelib.codecontext import CodeContext
+        from idlelib.paragraph import FormatParagraph
+        from idlelib.parenmatch import ParenMatch
+        from idlelib.rstrip import RstripExtension
+        from idlelib.runscript import ScriptBinding
```

Comment:
Why here instead with other class imports?  Is there a circular import problem?

Reply:
Travis or AppVeyor would error out with them when they were with the other imports, so I was pretty sure it was a circular import.

Reply:
I am sure that some of the imports, such as the one for paragraph, could be moved to module level.  They should be tested one by one, and a comment added for those that must stay.   I will eventually do this is you don't.  I hope to eventually reduce the loopy dependencies, so I want to know which fail otherwise.

Reply:
To do import just once, on first instance: at class level, add for instance 'AutoComplete = None' and in __init__: 'if AutoComplete == None: self.do_imports()' where do_imports does delayed imports and sets class attributes.